ivy_tests/test_ivy/helpers/hypothesis_helpers/array_helpers.py

```diff
@@ -1541,32 +1541,25 @@ def arrays_for_pooling(
     if not isinstance(data_format, str):
         data_format = draw(data_format)
     array_dim = x[0].ndim
-    if array_dim == 5:
-        kernel = draw(
-            st.tuples(
-                st.integers(1, in_shape[1]),
-                st.integers(1, in_shape[2]),
-                st.integers(1, in_shape[3]),
-            )
-        )
-    if array_dim == 4:
-        kernel = draw(
-            st.tuples(st.integers(1, in_shape[1]), st.integers(1, in_shape[2]))
-        )
-    if array_dim == 3:
-        kernel = draw(st.tuples(st.integers(1, in_shape[1])))
-    new_kernel = kernel
-    if return_dilation:
-        new_kernel = []
-        dilations = []
-        for i in range(len(kernel)):
+    kernel = []
+    new_kernel = []
+    dilations = []
+
+    for i in range(array_dim):
+        if i + 1 < len(in_shape):
+            kernel.append(draw(st.integers(1, in_shape[i+1])))
+        else:
+            kernel.append(draw(st.integers(1, 10)))  # Replace 10 with an appropriate maximum value
+
+        if return_dilation:
             if kernel[i] > 1:
-                max_dilation = (in_shape[i + 1] - kernel[i]) // (kernel[i] - 1) + 1
+                max_dilation = (in_shape[i+1] - kernel[i]) // (kernel[i] - 1) + 1
                 dilations.append(draw(st.integers(1, max_dilation)))
-                new_kernel.append(kernel[i] + (kernel[i] - 1) * (dilations[i] - 1))
+                new_kernel.append(kernel[i] + (kernel[i] - 1) * (dilations[i-1] - 1))
```

Comment:
Hello @mtnvdsk , can you please explain the rationale behind those changes? what is the bug/issue you're trying to fix?